packages/web-frontend/src/containers/Map/AreaTooltip.js

```diff
@@ -34,8 +41,82 @@ export class AreaTooltip extends Component {
     }
   }
 
+  getTextList() {
+    const {
+      orgUnitName,
+      hasMeasureValue,
+      measureOptions,
+      rawOrgUnitMeasureData,
+      orgUnitMeasureData,
+    } = this.props;
+
+    const defaultTextList = [
+      <Heading key={0} hasMeasureValue={hasMeasureValue}>
+        {orgUnitName}
+      </Heading>,
+    ];
+
+    const renderTextList = (data, textList = []) => {
+      Object.keys(data).forEach(key => {
+        textList.push(<span key={key}>{`${key}: ${data[key]}`}</span>);
+      });
+      return textList;
+    };
+
+    if (hasMeasureValue) {
+      const legacy = !measureOptions[0].notLegacy;
+      const { formattedMeasureData, onlyOneValue } = legacy
+        ? this.buildLegacyFormattedMeasureData(orgUnitMeasureData, measureOptions)
+        : this.buildFormattedMeasureData(rawOrgUnitMeasureData, measureOptions);
```

Comment:
Hmmm wouldn't it be possible for us to loop through the measureOptions here to work out which of the fields in the `orgUnitMeasureData` should be rendered? Rather than needing to pass in both `orgUnitMeasureData` and `rawOrgUnitMeasureData`? I think if this were the case we also would be able to avoid needing a `notLegacy` key right?

Reply:
Yeah I was thinking the same before. My concern is when we add a new config for `orgUnitMeasureData` we need to add those keys to those fields, and also `displayedValueKey` config will indicate another dynamic `key` of another config. Here `orgUnitMeasureData` and `rawOrgUnitMeasureData` can decouple the thing a bit since they are using sightly different logic. 

Eventually I guess we can refactor `orgUnitMeasureData` to be an extract config for measure data instead of adding configs in the list? @rohan-bes what do you think?